/*
 * Copyright (c) 2016 Cisco and/or its affiliates, and 
 *                    Cable Television Laboratories, Inc. ("CableLabs")
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at:
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import "HalCommon.proto";
package HalOperation;

//This message should be the first message for agent interface
message HalClientHello{
    required string MsgType = 1 [default= "HalClientHello"];

    //The Client ID
    required string ClientID = 10;
}

message HalClientHelloRsp {
    required string MsgType = 1 [default = "HalClientHelloRsp"];

    required string ClientID = 10;
}

//Hal configuration msg
message HalConfig {
    required string MsgType = 1 [default = "HalConfig"];

    optional string SrcClientID = 10;
    optional string DstClientID = 11;
    optional int32 SeqNum = 12;

    required int32 CfgMsgType = 20;
    required bytes CfgMsgPayload = 21;
}

message HalConfigRsp {
    required string MsgType = 1 [default = "HalConfigRsp"];

    optional string SrcClientID = 10;
    optional string DstClientID = 11;
    optional int32 SeqNum = 12;

    required HalCommon.Rsp Rsp = 20;

    required int32 CfgMsgType = 30;
    required bytes CfgMsgPayload = 31;
}

//Hal Notification Msg
message HalNotification {
    required string MsgType = 1 [default = "HalNotification"];
    required string ClientID = 10;
    required int32 HalNotificationType = 20;
    required bytes HalNotificationPayLoad = 21;
}

